### PR TITLE
Move Clipped Average Filter to C++

### PR DESCRIPTION
**Pull request by jeremykubica**
- Results in a >10x speed up in execution.
- Include logic to handle NO_DATA tags (which aren't passed by the python code yet)
- Add more tests to confirm the logic

### CODE REVIEW

File: search/src/Filtering.cpp [hunk 1]
```diff
@@ -46,6 +46,98 @@ std::vector<int> sigmaGFilteredIndices(const std::vector<float>& values, float s
...
+    sort(scores.begin(), scores.end());
+
+    // Find all the indices where:
+    // 1) LH > lower_lh_limit
+    // 2) They do not have one of the top 'num_clipped' likelihoods.
+    // 3) LH != NO_DATA
+    int first_valid = 0;
+    int last_valid = 0;
+    double lh_sum_good = 0.0;
+    for (int t = 0; t < num_times - num_clipped; ++t) {
```
Comment (cchris28) [inline search/src/Filtering.cpp:92]: did you consider using an iterator here? I am curious if it makes it any faster or if the compiler optimizes it to be equivalent.

Comment (jeremykubica) [reply search/src/Filtering.cpp:92]: I didn't benchmark that particular change.

File: search/src/Filtering.h [hunk 2]
```diff
@@ -9,6 +9,7 @@
 #ifndef FILTERING_H_
 #define FILTERING_H_
 
+#include <queue>
```
Comment (maxwest-uw) [inline search/src/Filtering.h:12]: maybe I'm missing something, but I don't think we're using a queue anywhere in these changes, is it necessary for anything else?

**Review by maxwest-uw — COMMENTED** [review]
one super small question :) everything else looks good

File: search/src/Filtering.h (hunk 2 above)
Comment (jeremykubica) [reply search/src/Filtering.h]: Good catch. That is a mistake. I originally added it, thinking I would use a queue like the python approach. But it turns out that it is easier to use a sorted array multiple times and just track indices. So I never ended up using the queue. I've removed the include.

**Review by maxwest-uw — APPROVED** [review]

**Pull request merged**